src/components/approval-type/ApprovalType.vue

```diff
@@ -0,0 +1,157 @@
+        <v-radio-group class="payment-group" v-model="approvalTypeSelected" @change="radioButtonChanged">
+          <!-- COURT ORDER radio button -->
+          <v-radio id="court-order-radio" class="mb-0"
+             :label="getRadioText(ApprovalTypes.VIA_COURT_ORDER)"
+             :value="ApprovalTypes.VIA_COURT_ORDER"
+          />
+          <v-form ref="courtNumRef" id="court-num-form" v-model="valid" class="mt-4 ml-8">
+            <v-text-field
+              id="court-order-number-input"
+              v-model="courtOrderNumber"
+              label="Court Order Number"
+              :rules="courtOrderNumRules"
+              :disabled="approvalTypeSelected === ApprovalTypes.VIA_REGISTRAR"
+              @input="courtOrderNumberChanged"
+              @update:error="emitValidationError($event)"
+              filled
+            />
+          </v-form>
+          <!-- REGISTRAR radio button -->
+          <v-radio v-if="!isCourtOrderOnly"
+             id="registrar-radio" class="mb-0 pt-2"
+             :label="getRadioText(ApprovalTypes.VIA_REGISTRAR)"
+             :value="ApprovalTypes.VIA_REGISTRAR"
+          />
+        </v-radio-group>
+      </v-col>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Emit, Prop } from 'vue-property-decorator'
+import { FormIF } from '@bcrs-shared-components/interfaces'
+
+@Component({})
+export default class ApprovalType extends Vue {
+  // Refs
+  $refs!: Vue['$refs'] & {
+    courtNumRef: FormIF
+  }
+
+  ApprovalTypes = {
+    VIA_COURT_ORDER: 'VIA COURT ORDER',
+    VIA_REGISTRAR: 'VIA REGISTRAR'
+  }
+
+  /** Draft court order number. */
+  @Prop({ default: '' }) readonly draftCourtOrderNumber!: string
+
+  /** Whether approved by the registrar. */
+  @Prop({ default: false }) readonly draftApprovedByRegistrar!: boolean
+
+  /** filing name used in radio options. */
+  @Prop({ default: 'restoration'}) readonly filingType!: string
+
+  /** Show only the court order option; remove via registrar option. */
+  @Prop({ default: false }) readonly isCourtOrderOnly!: boolean
+
+  // Local properties
+  private courtOrderNumber = ''
+  private courtOrderNumRules = []
+  private valid = false
+  private approvalTypeSelected = ''
+
+  /** Called when component is mounted. */
+  mounted (): void {
+    // Copy props to mutable properties
+    if (this.draftApprovedByRegistrar) {
+      this.courtOrderNumber = ''
+      this.approvalTypeSelected = this.ApprovalTypes.VIA_REGISTRAR
+    }
+    if (this.draftCourtOrderNumber) {
+      this.courtOrderNumber = this.draftCourtOrderNumber
+      this.approvalTypeSelected = this.ApprovalTypes.VIA_COURT_ORDER
+    }
+  }
+
+  /** Triggers the form validation. */
+  public validate (): boolean {
+    let status = this.$refs.courtNumRef.validate()
+    this.$emit('emitValid', status)
+    return status
+  }
+
+  private emitValidationError (event: boolean): void {
+    this.$emit('emitValid', !event)
```

Comment:
We usually use the `@Emit` decorator instead of this, but it's valid.

Reply:
Yes, but the `@Emit` decorator doesn't allow me to change the value of the event.  On line 22, I call `@update:error="emitValidationError($event)"`  If there's a better way to do this, please let me know.

Reply:
If you return the value in an emit method, it should pass the value along, eg.

```
@Emit('myEvent')
protected emitMyEvent (myEvent: string): string {
   return myEvent
}
```

Or even simpler syntax:
```
@Emit('myEvent') protected emitMyEvent (myEvent: string): void {}
```